src/mango_selector.erl

```diff
@@ -411,6 +411,18 @@ match({[{<<"$all">>, Args}]}, Values, _Cmp) when is_list(Values) ->
 match({[{<<"$all">>, _Args}]}, _Values, _Cmp) ->
     false;
 
+%% This is for $elemMatch and possibly $in because of our normalizer.
+%% A selector such as { "results": { "$elemMatch": { "$gte": 80, "$lt": 85 } } }
+%% gets normalized to:
+%% {[{<<"results">>,
```

Comment:
Also, could you clean up the indentation here?

    {[{<<"results">>,
        {[{<<"$elemMatch">>,
            {[{<<"$and">>, [
                {[{<<>>,{[{<<"$gte">>,80}]}}]},
                {[{<<>>,{[{<<"$lt">>,85}]}}]}
            ]}]}
        }]}
    }]}.